clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java

```diff
@@ -83,6 +85,15 @@ public void configure(Map<String, ?> configs) throws KafkaException {
         if (endpointIdentification != null)
             this.endpointIdentification = endpointIdentification;
 
+        String secureRandomImplementation = (String) configs.get(SslConfigs.SSL_SECURE_RANDOM_IMPLEMENTATION_CONFIG);
+        if (secureRandomImplementation != null) {
+            try {
+                this.secureRandomImplementation = SecureRandom.getInstance(secureRandomImplementation);
+            } catch (GeneralSecurityException e) {
```

Comment:
I've been thinking about this, and if it was anything other than security I'd agree with you immediately. Let's take a hypothetical situation where the SHA1PRNG implementation is found to have insufficient entropy to be considered secure. The JRE will maintain the defaults, but history tells us that the insecure implementation will not be removed (ciphers that are considered not secure are still available, they're just not the default). Someone using Kafka may think that everything is fine because they are using default behavior, whereas the default behavior of Kafka differs from the default behavior of the JRE. In addition, it may not even be clearly announced as SHA1PRNG is not the default choice.

Certainly we can issue a bugfix release for Kafka and call out the security issue, but then we will be responsible for making a change to Kafka's default behavior. We might have another PRNG choice that is available across the board that we can change to as the default. Or it might be that we have to shift back to the JRE default behavior, which will cause a severe performance regression. Or we might have to add a significant amount of logic to try and pick a PRNG implementation based on the platform.